### ISSUE TITLE
Extraneous error output

### ISSUE THREAD

**Issue by padraic-padraic** [LLNL/gtest-mpi-listener#2]
Thank you very much for this project, I'm using it to test an MPI interface to a scientific library I've been working on as part of my PhD.

I'm currently setting up `gtest-mpi-listener`, and in particular when I'm running the example tests I get a lot of error output that makes it hard to parse. Initially, before the tests run, there's several rounds of
```
MPI must be initialized before RUN_ALL_TESTS!
Add '::testing::InitGoogleTest(&argc, argv);
     MPI_Init(&argc, &argv);' to your 'main' function!
```
on the terminal, before tests start running normally.

Adding some Debug info, I can see this is happening during the environment setup. Sensibly, the setup calls `FAIL` if MPI is not initialised, but in normal usage what happens is Googletests calls setup over and over until MPI has finished initialising.

Interestingly, if I delete the MPI_INIT call from the tests, this error prints only once, and then the tests error later when other parts of the MPI API are called.

Edit: This was related to compiling tests in release mode. Compiling in debug mode, I now get the correct behaviour when setting up tests.

---

The second problem I had is that the call to `MPI_Comm_free` in `~MPIMinimalistPrinter` happens after the call to `MPI_Finalize` in `MPIEnvironment.TearDown`, and so the threads abort at the end after successfully running all the tests. This can be fixed by moving the code from `~MPIMinimalistPrinter` to the method `OnEnivronmentTearDownStart`.

**Comment by bryan-lunt** [comment]
Hi @padraic-padraic , not to plug my fork, but if you use mine, you can just use the JSON or XML output.

https://github.com/bryan-lunt-supercomputing/gtest-mpi-listener
